### PR TITLE
Add landmark roles and ARIA

### PR DESCRIPTION
**Pull request by antibland**
I liked your "stateless HTML" article and thought I'd make an accessibility contribution to it.

### CODE REVIEW

**Comment by enyo** [comment]
Thanks for that! I didn't include any because it was simply a proof of concept, but you're absolutely right: even in this case it should be there, because it's important. I have a small inline comment though.

File: _layouts/default.html [hunk 1]
```diff
@@ -12,24 +12,24 @@
...
 
 <header>
-  <nav>
-    <a href="index.html" {% if page.site == 'home' %}class="active"{% endif %}>Home</a>
-    <a href="contact.html" {% if page.site == 'contact' %}class="active"{% endif %}>Contact</a>
-    <a href="about.html" {% if page.site == 'about' %}class="active"{% endif %}>About</a>
+  <nav role="navigation">
+    <a aria-label="home page" href="index.html" {% if page.site == 'home' %}class="active"{% endif %}>Home</a>
+    <a aria-label="contact me" href="contact.html" {% if page.site == 'contact' %}class="active"{% endif %}>Contact</a>
+    <a aria-label="about me" href="about.html" {% if page.site == 'about' %}class="active"{% endif %}>About</a>
```
Comment (enyo) [inline _layouts/default.html]: It's not necessary to set `aria-label`, since the text in the `<a>` element is describing it anyway.

**Comment by antibland** [comment]
You're right about the labels. Removed.

**Comment by enyo** [comment]
In case you are wondering in the future: I removed the `#main` id and the `role="main"` attribute from the main content.
The `role="main"` is not necessary since that's already defined by the `<main>` element. The only reason to add it, would be for IE, since it doesn't have support for it yet. Since this is only a proof of concept though, I prefer not adding browser workaround fixes.

**Pull request merged**